aten/src/ATen/native/cuda/UnaryOpsKernel.cu

```diff
@@ -99,6 +99,15 @@ void sign_kernel_cuda(TensorIterator& iter){
     }
 }
 
+void sigmoid_kernel_cuda(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.dtype(), "sigmoid_cuda", [&]() {
+      gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
+         scalar_t one = scalar_t(1);
+         return  one / (one + std::exp(- a));
```

Comment:
Avoid using std:: functions in the device code, this can cause ROCm failures (see e.g. #25791 that had to be backed out)

Reply:
As comment on https://github.com/pytorch/pytorch/blob/7f89464b2d9b20d728c1d29c33d32ada51ab9937/aten/src/ATen/native/Distributions.h#L7, floor also cause ROCm failures, why std::floor is used in  https://github.com/pytorch/pytorch/blob/af96e0cb5bf83175cb648659ce3b40d84a807c77/aten/src/ATen/native/cuda/UnaryOpsKernel.cu#L43